PROTOCOL.md

```diff
@@ -1343,6 +1344,17 @@ When enabled:
  - The value of `delta.generationExpression` SHOULD be parsed as a SQL expression.
  - Writers MUST enforce that any data writing to the table satisfy the condition `(<value> <=> <generation expression>) IS TRUE`. `<=>` is the NULL-safe equal operator which performs an equality comparison like the `=` operator but returns `TRUE` rather than NULL if both operands are `NULL`
 
+## Default Columns
+
+Delta supports defining default expressions for columns on Delta tables. Delta will generate default values for columns when users do not explicitly provide values for them when writing to such tables. To enable Default Columns:
+- The table must be on Writer Version 4, or
+- The table must be on Writer Version 7, and a feature name `allowColumnDefaults` must exist in the table `protocol`'s `writerFeatures`.
+
+When enabled:
+- The `metadata` for the column in the table schema MAY contain the key `CURRENT_DEFAULT`.
```

Comment:
@felipepessoto You're right. This suggested key comes from the key of the same name in the Apache Spark implementation of this feature [1]. That's implemented in this file [2]. In this way the Delta implementation would become compatible with Apache Spark right away. We could change to a more consistent key like `delta.currentDefaultExpression`, but then to make it work with Spark, we'd have to first change Spark, then wait several months for the next Spark and Delta versions to release, and then update Delta after.

Brainstorming, we could also propose `delta.currentDefaultExpression` as the primary key and `CURRENT_DEFAULT` as a valid alias for purposes of this compatibility. Would this be OK with you?

[1] https://issues.apache.org/jira/browse/SPARK-38334

[2] https://github.com/apache/spark/blob/536439244593d40bdab88e9d3657f2691d3d33f2/sql/api/src/main/scala/org/apache/spark/sql/catalyst/util/ResolveDefaultColumnsUtils.scala

Reply:
I was not aware of the Spark metadata. Now I have more questions haha, if you can help me understand:

1-) Is this Spark metadata stored in Hive?
2-) If #1 answer is Yes. For Delta Tables it would be stored in both Hive and Delta Log? If so, then I'd use the same key name.
3-) Does Delta for Databricks support columns defaults? If so, do you know what keys they use? Probably better to use the same for compatibility.

Thanks

Reply:
1) Yes, it is stored in the column metadata in Hive Metastore.
2) Yes, it would also be stored in the Delta log, but only by nature of the column in the "metadata"*.
3) Yes, once the Spark support** was implemented, Databricks also included that support for CSV, JSON, Orc, Parquet, and Delta. All use the same metadata key name.

* https://github.com/delta-io/delta/blob/master/PROTOCOL.md#struct-field
** https://issues.apache.org/jira/browse/SPARK-38334

Reply:
Thanks!
I suggest keeping as is now, CURRENT_DEFAULT, as it is consistent with Spark and Databricks